src/main/java/org/elasticsearch/indices/memory/IndexingMemoryController.java

```diff
@@ -191,6 +192,11 @@ private int updateShardStatuses(EnumSet<ShardStatusChangeType> changes, List<Ind
                         continue;
                     }
 
+                    // nocommit: better solution
+                    if (indexShard instanceof ShadowIndexShard) {
```

Comment:
add a method to `IndexShard#canIndex() or #canWrite or #hasIndexer()` and make the decision based on that?